R/pensynth.R

```diff
@@ -73,30 +72,34 @@ pensynth <- function(X1, X0, v, lambda = 0, opt_pars = osqp::osqpSettings(polish
   X1VX0 <- crossprod(X1v, X0v)
   Delta <- apply(X0v - c(X1v), 2, crossprod)
 
-  # linear constraint matrix
-  Amat   <- rbind(rep(1, N_donors), diag(N_donors))
-  lbound <- c(1, rep(0, N_donors))
-  ubound <- rep(1, N_donors + 1)
+  # Constraint matrices
+  Amat <- rbind(
+    rep(1, N_donors), # Sum to 1 constraint
+    -diag(N_donors) # Individ. weights gte 0 constraint
+  )
+  B <- c(
+    1, # Sum to 1 constraint
+    rep(0, N_donors) # Individ. weights gte 0 constraint
+  )
 
-  # stop annoying printing with capture.output.
-  o <- capture.output({
+# Run the quadratic program solver
+  result <- clarabel(
```

Comment:
```suggestion
  result <- clarabel::clarabel(
```
This will fix the error!